### DISCUSSION TITLE
Kobra Neo part cooling fan not running under 100% is probably a controller issue and not a fan issue

### DISCUSSION THREAD

**Discussion by LawfulGreen** [1coderookie/KobraGoNeoInsights#10] [General]
As the title states, the issue described [here](https://1coderookie.github.io/KobraGoNeoInsights/hardware/printhead/#neo_6) is also present with a replacement fan ( [Olearn DC Brushless 5015 24V blower fan](https://3dprintingcanada.com/products/air-blower-fan-5015-24v)). From that data point I believe the issue might be controller-based, where the latest board revision (printer purchased in Canada in July 2023) might have problems with PWM fan control.

![](https://user-images.githubusercontent.com/136001967/265682741-556bb65b-4e1b-4533-994f-45742d1d02c3.jpg)

Has anyone else observed this behaviour?

**Comment by 1coderookie** [comment]
Yes, I'm aware of that and I'm currently investigating the problem (I'm just a bit short on time right now..). 
Based on my findings so far it isn't a controller problem though, it's a problem of certain fans not running (well) with PWM / lower voltages. Really short summary: PWM works in a way that the fan will be switched on and off quickly, which results in a lower overall voltage so to say. Not all 24V fans run at lower voltages than 24V though, and that seems to be the problem.  
I'll get back to you once I did further tests.  
In the meantime: could you maybe post a picture of the label of the fan here, so that we can collect which ones work and which ones don't?

**Reply by LawfulGreen** [reply]
![20230904_153704](https://github.com/1coderookie/KobraGoNeoInsights/assets/136001967/556bb65b-4e1b-4533-994f-45742d1d02c3)

**Reply by 1coderookie** [reply]
Great, thx.  
EDIT: can you maybe edit the post with the pic and add the link to where you got it? Just to have somewhat like a list which ones to avoid.. 

I personally also have one cheap noname 5015 which doesn't work with PWM.  
@ProXinous just got a cheap 5015 tho which does work.  

Maybe we can create somewhat like a collection here of fans which do and which don't work with the PWM..

**Reply by ProXinous** [reply]
Hey,  although I had no time to further test different frequencies, I can confirm that my 5015 fan which was 2$, actually does work well when PWM controlled (20khz, without any kickstart 0-100%) although I'm pretty sure it's not the desired frequency since it's still pushing lots of air even on 1%. But still compared to stock HSC one  - much more control (not even saying about how quiet it is)
![IMG_20230905_134939](https://github.com/1coderookie/KobraGoNeoInsights/assets/142351534/181c9adf-9320-4bda-be08-c168434e29a2)

**Reply by 1coderookie** [reply]
Great, thx for the feedback :)

**Reply by LawfulGreen** [reply]
@ProXinous could you please link where you got that fan?

**Reply by 1coderookie** [reply]
@LawfulGreen you probably didn't see my edit above: could you maybe edit the post with the pic and add the link to where you got it? Just to have somewhat like a list which ones to avoid.. TIA

**Reply by LawfulGreen** [reply]
It's updated

**Reply by ProXinous** [reply]
Hey, sorry for late response. I doubt my link would give you anything unless you're Polish since I got mine on some random auction on Allegro.pl which is equivalent of eBay but specifically for Poland. It was described as generic 24v 5015 blower fan for 3d printer. So buying cheap fans is no different than Russian roulette ;P

**Comment by LawfulGreen** [comment]
I'm coming back to this, in case it's useful to people (this could be added to the hotend mods guide section?):

I've done a bunch of experimentation with the 5015 fan from the OP, as well as a Creality branded (CHA5024RL-15B), I got as a replacement hoping it would solve the issue.
What I found is that the main problem seems to be with the PWM frequency. I'm running Klipper and using a printer.cfg file credited to xpeng ([printer_cfg.txt](https://github.com/1coderookie/KobraGoNeoInsights/files/13846000/printer_cfg.txt)) where active frequency is 20kHz. This is waaay too high, and it's what's stopping the fans from working under 95%.
With a frequency of less than 1kHz, the fans start to work from 10%, although that seems to cause the fan to vibrate with the PWM frequency regardless of power setting (except for 100%), with both fans I have whining loudly for settings above 70-80Hz.

I tried dozens of values to gauge resonance and vibration transmission to the rest of the printhead (by feel, I don't have access to accelerometers to validate empirically), and in the end I settled on using a cycle time value of $1/({x} * \frac{255}{60})$ equivalent to frequencies between 8Hz and 40Hz (mine works best at 17Hz, $x = 4$.) The rest of the formula is driven by the fact that we are changing the fan RPM with Gcode that has only 255 levels, so the PWM resolution is an integer multiple of 255 (the larger, the better, apart from vibration issues)
It's arbitrary, for sure, but it works. 

Let me know if this has worked for you!

**Reply by 1coderookie** [reply]
Thanks for sharing your findings!   

The problem with this is tho that it's pretty much dependant on the actual fan model.  
Generally speaking, some fans are working better with lower frequencies, some with higher frequencies, some apparently don't work well with PWM at all.   
So unfortunately there's no specific setting one could recommend in general, the only recommendation could be to test different frequencies for the specific fan one has gotten (~~what I have add as a note to the part cooling fan section~~ - thanks for the reminder! -> done ;) ).   

Just as a note fyi about the stock setting/set up:  
The specific fan model which was originally being used at the Neo as a part cooling fan (the Coolcox) seemed to run best at the high 20kHz AC has set in the fw. In the beginning when Klipper became available for the Neo but when it wasn't known about the 20kHz setting, several users reported dying fans due to the default 100Hz setting Klipper was using.  
After investigating the fw further, xpeng (iirc) found out that it's actually 20kHz AC has set in the fw. That's why it was changed in the first place, and after doing so no more dying fans have been reported.  
I personally tested different frequencies back then, and I can confirm that 'weird' and noisy behaviour occured with significantly different settings and that 20kHz does seem to work best *for that specific model*.  
Then AC changed the type of part cooling fan and used that HSC model, but they didn't change anything in the fw. As that new type doesn't really work with PWM (since then, AC support tells customers that it's normal that a part cooling fan only work in an on/off state btw..), I then tested various frequencies from 100Hz (iirc - might be that I also tried lower frequencies, I really don't remember anymore..) up to the stock 20kHz. And while I can say that it got a little bit better with low frequencies, I wasn't able to get the PWM working completely fine with that model.   

Same with a cheap 5015 no-name blower I once got: no matter which PWM frequency I use, that one doesn't work at all with PWM.. :(

**Reply by jaltgen** [reply]
Thanks for the details on this. Since using Klipper, my parts fan died. Assuming this is due to the 100Hz default in Klipper. I now got an aftermarket fan to fit, and ran this also at 100Hz, assuming it wasn't a weird 20kHz model that would do PWM. I'm using a BTT SRT 3 mini board. The fan motor was super hot to the touch when it stopped working, and would only wiggle back and forth. On switching off the fan, it then jerked and spun down. Testing later if it works again at 20kHz maybe?
